geode-redis/src/main/java/org/apache/geode/redis/internal/executor/RedisKeyCommandsFunctionExecutor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.executor;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.redis.internal.ByteArrayWrapper;
+import org.apache.geode.redis.internal.RedisCommandType;
+import org.apache.geode.redis.internal.RedisData;
+
+public class RedisKeyCommandsFunctionExecutor implements RedisKeyCommands {
+  private Region<ByteArrayWrapper, RedisData> region;
+
+  public RedisKeyCommandsFunctionExecutor(
+      Region<ByteArrayWrapper, RedisData> region) {
+    this.region = region;
+  }
+
+  @Override
+  public boolean del(ByteArrayWrapper key) {
+    return (boolean) CommandFunction.execute(RedisCommandType.DEL, key, new Object[] {}, region);
```

Comment:
Something I find slightly confusing on CommandFunction is that is has a static method named execute. When looking at a Function I tend to look for its "instance" method named execute. Something we could consider is having all our *FunctionExecutor classes, subclass a common class that has the code from the static CommandFunction execute. I think that could also simplify how much we need to pass to this method (we would no longer need to pass the region since we could access it from the instance). Let me know what you think. No need to do this on this PR

Reply:
yea, the static execute method looks a little weird to me as well.
I like the idea of having a simple interface other than the FunctionService to expose the function call, but the name `execute` is definitely confusing.
Some kind of code sharing in the *FunctionExecutor classes is a good idea.